### PR TITLE
[doc] update posix app README

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR updates the doc for using posix app. Basically, it updates
the instructions to use cmake build posix core and transceiver.

### CODE REVIEW

File: src/posix/README.md [hunk 1]
```diff
@@ -16,70 +16,98 @@ The figure below shows the architecture of OpenThread running in transceiver mod
...
+#### Clone the repo
 
 ```sh
-make -f examples/Makefile-nrf52840
-arm-none-eabi-objcopy -O ihex output/nrf52840/bin/ot-rcp ot-rcp.hex
-nrfjprog -f nrf52 --chiperase --reset --program ot-rcp.hex
-nrfjprog -f nrf52 --pinresetenable
-nrfjprog -f nrf52 --reset
+# `cd` to your directory to clone the repo
+git clone git@github.com:openthread/ot-nrf528xx.git
```
Comment (bukepo) [inline src/posix/README.md:56]:
```suggestion
git clone https://github.com/openthread/ot-nrf528xx.git
```

File: src/posix/README.md [hunk 2]
```diff
@@ -16,70 +16,98 @@ The figure below shows the architecture of OpenThread running in transceiver mod
          POSIX                                          Chip
 ```
 
-## Build
+## Build POSIX core
```
Comment (bukepo) [inline src/posix/README.md:19]:
```suggestion
## Build POSIX CLI
```

File: src/posix/README.md [hunk 3]
```diff
@@ -88,11 +116,11 @@ OpenThread Posix Daemon mode uses a unix socket as input and output, so that Ope
 
 ```
 # build daemon mode core stack for POSIX
-make -f src/posix/Makefile-posix DAEMON=1
+./script/cmake-build posix -DOT_DAEMON=1
```
Comment (bukepo) [inline src/posix/README.md:119]:
```suggestion
./script/cmake-build posix -DOT_DAEMON=ON
```

**Review by bukepo — APPROVED** [review]

File: src/posix/README.md [hunk 4]
```diff
@@ -16,70 +16,98 @@ The figure below shows the architecture of OpenThread running in transceiver mod
...
 
-## Test
+If built successfully, the binary should be found at: `build/posix/src/posix/ot-cli`.
 
-**NOTE** Assuming the build system is 64bit Linux, you can use the normal OpenThread CLI as described in the [command line document](../../src/cli/README.md). You can also perform radio diagnostics using the command [diag](../../src/core/diags/README.md).
+## Transceivers on different platforms
+
+### Simulation
+
+OpenThread provides an implemenation on simulation platform which enables running a simulated transceiver on the host.
```
Comment (Vyrastas) [inline src/posix/README.md:31]: on the simulation platform

File: src/posix/README.md [hunk 5]
```diff
@@ -16,70 +16,98 @@ The figure below shows the architecture of OpenThread running in transceiver mod
...
-make -f examples/Makefile-simulation
-./output/posix/bin/ot-cli 'spinel+hdlc+forkpty://output/simulation/bin/ot-rcp?forkpty-arg=1'
+./build/posix/src/posix/ot-cli 'spinel+hdlc+forkpty://build/simulation/examples/apps/ncp/ot-rcp?forkpty-arg=1'
 ```
 
-### With Real Device
+### nRF52840
 
-#### nRF52840
+The nRF example platform driver has been moved out of OpenThread repo and is placed in a separate repo: [ot-nrf528xx](https://github.com/openthread/ot-nrf528xx). To build transceivers on nRF platform, we need to clone that repo.
```
Comment (Vyrastas) [inline src/posix/README.md:50]: I don't think we need to say the example has moved, let's just state where it is now.

"The nRF example platform driver can be found in the [ot-nrf528xx](https://github.com/openthread/ot-nrf528xx) repo. To build transceivers on the nRF platform, we must clone that repo."

**Review by Vyrastas — APPROVED** [review]
Just a few minor comments

**Comment by Irving-cl** [comment]
@LuDuda Could you take a look when you have a chance :)

**Review by LuDuda — COMMENTED** [review]
Thank you @Irving-cl for doing this!

This PR looks good. However, I have recently created a little more extended documentation to our SDK on how to build RCP, let me create a PR to your brach, so you can decide if it's worth adding additional information.

**Comment by Irving-cl** [comment]
@LuDuda

That's great! Thanks.

**Comment by LuDuda** [comment]
@Irving-cl I just filed https://github.com/Irving-cl/openthread/pull/2 which you may want to fetch into the PR if it's not too detailed.

**Comment by Irving-cl** [comment]
@googlebot I consent.

**Comment by Irving-cl** [comment]
@LuDuda 

I directly merged your PR in this branch. Please comment `@googlebot I consent.` in this PR.

**Comment by LuDuda** [comment]
@googlebot I consent

**Review by LuDuda — APPROVED** [review]

**Comment by Irving-cl** [comment]
@jwhui I fixed it.

**Review by jwhui — APPROVED** [review]

**Pull request merged**